jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -9,825 +9,181 @@
+    else:
+        raise NotImplementedError("Transaction size estimation not" +
+                                  "yet implemented for type: " + txtype)
 
 def pubkey_to_p2pkh_script(pub, require_compressed=False):
-    """ Construct a pay-to-pubkey-hash scriptPubKey
-    given a single pubkey. Script returned in binary.
-    The require compressed flag may be used to disallow
-    uncompressed keys, e.g. in constructing a segwit
-    scriptCode field.
     """
-    if not isinstance(pub, bytes):
-        pub = binascii.unhexlify(pub)
-    return pubkey_to_script(pub, P2PKH_PRE, P2PKH_POST)
-
-def pubkey_to_p2sh_p2wpkh_script(pub):
-    """ Construct a nested-pay-to-witness-pubkey-hash
-    scriptPubKey given a single pubkey.
-    Script returned in binary.
+    Given a pubkey in bytes, return a CScript
+    representing the corresponding pay-to-pubkey-hash
+    scriptPubKey.
     """
-    if not isinstance(pub, bytes):
-        pub = binascii.unhexlify(pub)
-    wscript = pubkey_to_p2wpkh_script(pub)
-    return P2SH_P2WPKH_PRE + bin_hash160(wscript) + P2SH_P2WPKH_POST
-
-def pubkey_to_p2sh_p2wpkh_address(pub, magicbyte=5):
-    """ Construct a nested-pay-to-witness-pubkey-hash
-    address given a single pubkey; magicbyte defines
-    network as for any p2sh address.
-    """    
-    if not isinstance(pub, bytes):
-        pub = binascii.unhexlify(pub)
-    script = pubkey_to_p2wpkh_script(pub)
-    return p2sh_scriptaddr(script, magicbyte=magicbyte)
+    if not is_valid_pubkey(pub, require_compressed=require_compressed):
+        raise Exception("Invalid pubkey")
+    return CScript([OP_DUP, OP_HASH160, Hash160(pub),
+                                          OP_EQUALVERIFY, OP_CHECKSIG])
 
 def pubkey_to_p2wpkh_script(pub):
-    """ Construct a pay-to-witness-pubkey-hash
-    scriptPubKey given a single pubkey. Note that
-    this is the witness program (version 0). Script
-    is returned in binary.
     """
-    if not isinstance(pub, bytes):
-        pub = binascii.unhexlify(pub)
-    return pubkey_to_script(pub, P2WPKH_PRE,
-                            require_compressed=True)
-
-def pubkey_to_p2wpkh_address(pub):
-    """ Construct a pay-to-witness-pubkey-hash
-    address (bech32) given a single pubkey.
-    """    
-    script = pubkey_to_p2wpkh_script(pub)
-    return script_to_address(script)
-
-def pubkeys_to_p2wsh_script(pubs):
-    """ Given a list of N pubkeys, constructs an N of N
-    multisig scriptPubKey of type pay-to-witness-script-hash.
-    No other scripts than N-N multisig supported as of now.
+    Given a pubkey in bytes (compressed), return a CScript
+    representing the corresponding pay-to-witness-pubkey-hash
+    scriptPubKey.
     """
-    N = len(pubs)
-    script = mk_multisig_script(pubs, N)
-    return P2WSH_PRE + bin_sha256(binascii.unhexlify(script))
+    if not is_valid_pubkey(pub, True):
+        raise Exception("Invalid pubkey")
+    return CScript([OP_0, Hash160(pub)])
```

Comment:
Can also be done with `P2WPKHCoinAddress.from_pubkey(pub).to_scriptPubKey()`

Reply:
Addressed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/1e2fc3848295a05f4ee685fd435f277c378acadc